pkgs/by-name/da/daytona-bin/package.nix

```
@@ -0,0 +1,68 @@
+{ stdenvNoCC
+, lib
+, fetchurl
+, callPackage
+, makeWrapper
+}:
+
+let
+  dayton-bin-provider-docker = callPackage ./providers/docker { };
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "daytona-bin";
+  version = "0.9.0";
+
+  src =
+    let
+      urls = {
+        "x86_64-linux" = {
+          url = "https://download.daytona.io/daytona/v${finalAttrs.version}/daytona-linux-amd64";
+          hash = "sha256-vJVGFmaGP9oCCzdvhuAPsoTaxzGvdDKDupMYuepRUCA=";
+        };
+        "x86_64-darwin" = {
+          url = "https://download.daytona.io/daytona/v${finalAttrs.version}/daytona-darwin-amd64";
+          hash = "sha256-R63AQVt5DudzJub+TYcJiHkBGVeOhjvgJZgnqvJb8t0=";
+        };
+        "aarch64-linux" = {
+          url = "https://download.daytona.io/daytona/v${finalAttrs.version}/daytona-linux-arm64";
+          hash = "sha256-98OEhJ1gakPTVO73M9WW0QuSDgR42gNjoioEkkNbf6w=";
+        };
+        "aarch64-darwin" = {
+          url = "https://download.daytona.io/daytona/v${finalAttrs.version}/daytona-darwin-arm64";
+          hash = "sha256-YmLyioFueEfi/2Q+JwINDhkwo617/KUZrimz9CibdA8=";
+        };
+      };
+    in
+    fetchurl urls."${stdenvNoCC.hostPlatform.system}";
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    dayton-bin-provider-docker
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin $out/share/daytona/providers
+    cp -r $src $out/bin/daytona
+    chmod +x $out/bin/daytona
+    cp ${dayton-bin-provider-docker}/bin/daytona-docker-provider $out/share/daytona/providers/
```

IMO this is a a bit clearer.
```suggestion
    install -Dm755 $src $out/bin/daytona
    install -Dm755 ${daytona-bin-provider-docker}/bin/daytona-docker-provider -t $out/share/daytona/providers/
```